google-cloud-core/src/main/java/com/google/cloud/Binding.java

```diff
@@ -16,64 +16,94 @@
 
 package com.google.cloud;
 
+import static com.google.common.base.Predicates.in;
+import static com.google.common.base.Predicates.not;
+
 import com.google.api.core.BetaApi;
 import com.google.auto.value.AutoValue;
 import com.google.common.base.Predicate;
-import com.google.common.base.Predicates;
 import com.google.common.collect.Collections2;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import javax.annotation.Nullable;
 
+/**
+ * Class for Identity and Access Management (IAM) policies. IAM policies are used to specify access
+ * settings for Cloud Platform resources. A policy is a list of bindings. A binding assigns a set of
+ * identities to a role, where the identities can be user accounts, Google groups, Google domains,
+ * and service accounts. A role is a named list of permissions defined by IAM.
+ *
+ * @see <a href="https://cloud.google.com/iam/docs/reference/rest/v1/Policy">Policy</a>
+ */
 @BetaApi("This is a Beta API is not stable yet and may change in the future.")
 @AutoValue
 public abstract class Binding {
+  /** Get IAM Policy Binding Role */
   public abstract String getRole();
 
+  /** Get IAM Policy Binding Members */
   public abstract ImmutableList<String> getMembers();
 
+  /** Get IAM Policy Binding Condition */
   @Nullable
   public abstract Condition getCondition();
 
+  /** Create a Binding.Builder from an existing Binding */
   public abstract Builder toBuilder();
 
+  /** Create a new Binding.Builder */
   public static Builder newBuilder() {
-    return new AutoValue_Binding.Builder();
+    List<String> emptyMembers = ImmutableList.of();
+    return new AutoValue_Binding.Builder().setMembers(emptyMembers);
   }
 
   @AutoValue.Builder
   public abstract static class Builder {
+    /**
+     * Set IAM Role for Policy Binding
+     *
+     * @throws NullPointerException if the role is null.
+     */
     public abstract Builder setRole(String role);
 
-    public abstract Builder setMembers(List<String> members);
+    /**
+     * Set IAM Members for Policy Binding
+     *
+     * @throws NullPointerException if a member is null.
```

Comment:
You don't really need to repeat this everywhere. It's better to use `@Nullable` for the hopefully few cases where null is accepted.